### ISSUE TITLE
`typeof(fn())` when fn returns void should be legal

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#14610]
`typeof(fn())` when fn returns void should be legal

### Example
```nim
when true:
  proc fn = discard
  type T1 = void # ok
  type T2 = typeof(fn()) # bug: Error: expression 'fn()' has no type (or is ambiguous)
```

### Current Output
```
Error: expression 'fn()' has no type (or is ambiguous)
```

### Expected Output
```
works (and T2 is void)
```

### Additional Information
```
devel 1.3.5 30c09e460778099633db084532672970954bb328
```
this causes un-necessary workarounds in generic code

**Comment by mratsim** [comment]
Duplicate: https://github.com/nim-lang/Nim/issues/7397

RFC: https://github.com/nim-lang/RFCs/issues/508

(unless we want separate codepaths for type and typeof?)

**Comment by timotheecour** [comment]
ok; it's indeed dup of nim-lang/RFCs#508 ; however I think nim-lang/RFCs#508 can be fixed without having to do RFC nim-lang/RFCs#508
> (unless we want separate codepaths for type and typeof?)

no, I didn't intend that
